### PR TITLE
[native] Update CI dependency image to latest which is Centos 9 based

### PR DESCRIPTION
**Pull request by czentgr**
## Description
Updating the dependency image that was built with the Centos9 updates from Velox and Prestissimo.

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by czentgr** [comment]
@majetideepak FYI

**Comment by majetideepak** [comment]
@czentgr CI is failing. Looks like we need to make a change to the circle ci yml file with the new gcc tool chain path.

**Comment by czentgr** [comment]
Not sure what is happening with the circleci for linux-build-and-unit-test. When clicking on details it displayed a page saying "Something Unexpected Happened". In the banner is says operational.
@assignUser any idea what is going on?

**Comment by majetideepak** [comment]
@czentgr You need to make this change https://github.com/prestodb/presto/issues/22901#issuecomment-2146158079

File: .circleci/continue_config.yml [hunk 1]
```diff
@@ -131,6 +131,7 @@ jobs:
             - run:
                 name: 'Run Unit Tests'
                 command: |
+                  export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib:/usr/local/lib64"
```
Comment (czentgr) [inline .circleci/continue_config.yml:134]: Another option is to update the dockerfile (to make sure the paths are picked up like we do for the runtime image) for the dependency image and rebuild it.

Comment (majetideepak) [reply .circleci/continue_config.yml:134]: I think we should also install the dependencies in a custom directory using `INSTALL_PREFIX` and use the ldconfig approach.
Let's merge this and use a follow-up PR?

**Review by majetideepak — APPROVED** [review]

File: .circleci/continue_config.yml (hunk 1 above)
Comment (czentgr) [reply .circleci/continue_config.yml:134]: Sure.

Comment (majetideepak) [reply .circleci/continue_config.yml:134]: We can also avoid this issue by linking proxygen statically. The proxygen CMake recommends that.

**Comment by majetideepak** [comment]
@rschlussel can you please help by approving this PR? Thank you!

**Review by rschlussel — APPROVED** [review]
haven't looked into anything. approving based on @majetideepak's review.

**Comment by majetideepak** [comment]
Thanks, @rschlussel!

**Pull request merged**